File: android/DFM/src/main/java/com/dontflymoney/baseactivity/ResultHandler.java
package com.dontflymoney.baseactivity;

import com.dontflymoney.api.Step;
import com.dontflymoney.userdata.Language;
import com.dontflymoney.view.R;

import org.json.JSONException;
import org.json.JSONObject;

class ResultHandler
{
	private SmartActivity activity;
	private Navigation navigation;
	
	ResultHandler(SmartActivity activity, Navigation navigation)
	{
		this.activity = activity;
		this.navigation = navigation;
	}
	
	void HandlePostResult(JSONObject result, Step step)
	{
		try
		{
			if (result.has("error"))
			{
				String error = result.getString("error");
				
				activity.getMessage().alertError(error);

				if (error.contains(activity.getString(R.string.uninvited))
					|| error.contains("uninvited"))
				{
					navigation.logout();
				}				
			}
			else
			{
				JSONObject data = result.getJSONObject("data");
				
				if (data.has("Language"))
					Language.ChangeAndSave(activity, data.getString("Language"));
				
				activity.HandleSuccess(data, step);
			}
		}
		catch (JSONException e)
		{
			activity.getMessage().alertError(R.string.error_activity_json, e);
		}
	}

	void HandlePostError(String error, Step step)
	{
		activity.getMessage().alertError(error);
	}
	
	
}

File: android/DFM/src/main/kotlin/com/darakeon/dfm/activities/base/ResultHandler.kt
package com.darakeon.dfm.activities.base

import com.darakeon.dfm.R
import com.darakeon.dfm.activities.objects.SmartStatic
import com.darakeon.dfm.api.Step
import com.darakeon.dfm.user.Language
import com.darakeon.dfm.user.Theme
import org.json.JSONObject

internal class ResultHandler<T : SmartStatic>(private val activity: SmartActivity<T>, private val navigation: Navigation<T>) {

	fun HandlePostResult(result: JSONObject, step: Step) {
		if (result.has("error")) {
			val error = result.getString("error")

			activity.message.alertError(error)

			if (error.contains(activity.getString(R.string.uninvited)) || error.contains("uninvited")) {
				navigation.logout()
			}
		} else {
			val data = result.getJSONObject("data")

			if (data.has("Language"))
				Language.ChangeAndSave(activity, data.getString("Language"))

			if (data.has("Theme"))
				Theme.ChangeAndSave(activity, data.getString("Theme"))

			activity.HandleSuccess(data, step)
		}
	}

	fun HandlePostError(error: String) {
		activity.message.alertError(error)
	}


}
